emcc.py

```diff
@@ -831,6 +831,7 @@ def process_dynamic_libs(dylibs, lib_dirs):
   for dylib in dylibs:
     exports = webassembly.get_exports(dylib)
     exports = set(e.name for e in exports)
+    exports = [utils.removeprefix(e, '__em_js__') for e in exports]
```

Comment:
Why remove this prefix? A comment might be good as it looks surprising here I think.

Also, it seems like keeping the prefix might avoid the risk of a conflict with something else?

Reply:
Added a comment.

The point of this code here is simply to decide which symbols are allowed to be undefined in the main module (i.e. which symbols will be provided by the side module).   Duplicates here are not going to be problem, but I'm actually not sure what happens in the normal (non-shared) case when a symbol is exported both as a normal function and as a EM_JS function... whatever the case that would be an existing bug.